src/components/Sections/TeaserSection/index.tsx

```diff
@@ -6,54 +6,75 @@ import Headline from "@/components/Headline";
+import {
+  TeaserSectionEventsWithOn,
+  TeaserSectionProps,
+} from "@/types/sections";
 
 @Component({
   name: "TeaserSection",
 })
-class TeaserSection extends BaseComponent<TeaserSectionProps> {
+class TeaserSection extends BaseComponent<
+  TeaserSectionProps,
+  TeaserSectionEventsWithOn,
+  {
+    headline?: TeaserSectionProps["headline"];
+    kicker?: TeaserSectionProps["kicker"];
+    text?: TeaserSectionProps["text"];
+    button?: TeaserSectionProps["buttonText"];
+  }
+> {
   @Prop({ required: true }) headline!: TeaserSectionProps["headline"];
   @Prop({ required: true }) kicker!: TeaserSectionProps["kicker"];
   @Prop({ required: true }) text!: TeaserSectionProps["text"];
   @Prop({ required: false }) buttonText: TeaserSectionProps["buttonText"];
-  @Prop({ required: false })
-  handleButtonClick: TeaserSectionProps["handleButtonClick"];
-  @Prop({ required: false })
-  image: TeaserSectionProps["image"];
+  @Prop({ required: false }) image: TeaserSectionProps["image"];
 
   render() {
     return (
       <div class="w-full bg-gray-100 text-black">
         <Container>
           <Layout wrap>
             <LayoutItem width="full" lg={{ width: "5/12" }}>
-              <Headline
-                as="h3"
-                size="lg"
-                weight="light"
-                data-testid={"teasersection-kicker"}
-              >
-                {this.kicker}
-              </Headline>
-              <Headline size="xl" data-testid={"teasersection-headline"}>
-                <RichText content={this.headline} inline />
-              </Headline>
-              <RichText
-                class="text-sm"
-                data-testid={"teasersection-text"}
-                content={this.text}
-              />
-              {this.buttonText && (
-                <Button
-                  data-testid={"teasersection-button"}
-                  variant="animated"
-                  handleClick={() =>
-                    this.handleButtonClick && this.handleButtonClick()
-                  }
+              {this.$scopedSlots.kicker ? (
+                this.$scopedSlots.kicker(this.kicker)
+              ) : (
+                <Headline
+                  as="h3"
+                  size="lg"
+                  weight="light"
+                  data-testid={"teasersection-kicker"}
                 >
-                  {this.buttonText}
-                </Button>
+                  {this.kicker}
+                </Headline>
+              )}
+              {this.$scopedSlots.headline ? (
+                this.$scopedSlots.headline(this.headline)
+              ) : (
+                <Headline size="xl" data-testid={"teasersection-headline"}>
+                  <RichText content={this.headline} inline />
+                </Headline>
+              )}
+              {this.$scopedSlots.text ? (
+                this.$scopedSlots.text(this.text)
+              ) : (
+                <RichText
```

Comment:
Please remove the RichText Component and replace it with a simple div. (The RichText-Component is deprecated and will be removed in the next release)